components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -22,18 +22,47 @@ import java.util.UUID
  * on the API.
  */
 interface PushConnection : Closeable {
-    suspend fun subscribe(channelId: String, scope: String = ""): SubscriptionResponse
-    suspend fun unsubscribe(channelId: String): Boolean
+    /**
+     * Creates a push subscription for the given scope.
+     */
+    suspend fun subscribe(scope: PushScope, appServerKey: AppServerKey? = null): AutoPushSubscription
+
+    /**
+     * Un-subscribes a push subscription for the given scope.
+     */
+    suspend fun unsubscribe(scope: PushScope): Boolean
+
+    /**
+     * Un-subscribes all push subscriptions.
+     */
     suspend fun unsubscribeAll(): Boolean
+
+    /**
+     * Updates the registration token to the native Push API if it changes.
+     */
     suspend fun updateToken(token: String): Boolean
+
+    /**
+     * Checks validity of current push subscriptions.
+     *
+     * @return true if a push subscription was updated and a subscriber should be notified of the change.
```

Comment:
This will change to a `Map<PushScope, Boolean>`, right? Surely we don't want to notify _everyone_ if just one subscription is updated? If so, can you add a link to a github follow-up here?

Reply:
> Surely we don't want to notify everyone if just one subscription is updated?

In the case that we have a single subscription updated, we will be notifying everyone. Changing this also involves re-introducing the MessageBus.